src/components/Menu/Menu.tsx

```diff
@@ -147,6 +159,7 @@ class Menu extends AutoControlledComponent<Extendable<IMenuProps>, any> {
           vertical,
           index,
           active: parseInt(activeIndex, 10) === index,
+          submenuOpen: this.state.activeIndex === index ? this.state.submenuOpen : false,
```

Comment:
Always destructure state and pros, see line 150:

```jsx
     const { activeIndex } = this.state
```

Here you would add the keys you want to use.  `activeIndex` is already destructured in scope, so you can use it directly.  `submenuOpen` needs to be added to the destructured values.  Destructuring keeps code clean, but more importantly it also breaks object references.